runtime/compiler/optimizer/InlinerTempForJ9.cpp

```diff
@@ -866,80 +831,155 @@ TR_J9InlinerPolicy::createAnchorNodesForUnsafeGetPut(TR::TreeTop* treeTop,
+
+   TR::TreeTop *firstComparisonTree;
+   TR::TreeTop *ifTree;
+   TR::TreeTop *elseTree;
+
+   TR_OpaqueClassBlock *javaLangClass = comp()->getClassClassPointer(/* isVettedForAOT = */ true);
+
+   // Determine overall layout of IL - which test to place first (null test of object or
+   // setting of low-order bit of offset) - and whether direct or indirect access IL will
+   // be placed on the fall-through path of the first branch.
+   //
+   if (conversionNeeded)
+      {
+      firstComparisonTree = nullComparisonTree;
+      ifTree = arrayDirectAccessTreeTop;
+      elseTree = indirectAccessTreeTop;
+      }
+   else
+      {
+      firstComparisonTree = (javaLangClass != NULL) ? lowTagCmpTree : nullComparisonTree;
+      ifTree = indirectAccessTreeTop;
+      elseTree = directAccessTreeTop;
+      }
+
+   TR::Block *joinBlock =
+         createUnsafeGetPutCallDiamond(
+               callNodeTreeTop, firstComparisonTree, ifTree, elseTree);
+
+   debugTrace(tracer(), "\t In genCodeForUnsafeGetPut, joinBlock is %d\n", joinBlock->getNumber());
+
+   TR::Block *beforeCallBlock = prevTreeTop->getEnclosingBlock();
+
+   TR::Block *arrayDirectAccessBlock = NULL;
    TR::Block *indirectAccessBlock;
-   TR::Block * directAccessBlock;
+   TR::Block *directAccessBlock;
+
    if (conversionNeeded)
       {
+      indirectAccessBlock = beforeCallBlock->getNextBlock();
+
       //Generating block for direct access
-      indirectAccessBlock = nullComparisonBlock->getNextBlock();
       directAccessBlock = TR::Block::createEmptyBlock(lowTagCmpTree->getNode(), comp(),
-      indirectAccessBlock->getFrequency());
+                                                      indirectAccessBlock->getFrequency());
       directAccessBlock->append(directAccessTreeTop);
-      directAccessBlock->append(TR::TreeTop::create(comp(),
+      directAccessBlock->append(
+            TR::TreeTop::create(comp(),
                                 TR::Node::create(directAccessTreeTop->getNode(),
-                                TR::Goto, 0, joinBlock->getEntry())));
+                                                 TR::Goto, 0, joinBlock->getEntry())));
+
+      arrayDirectAccessBlock = firstComparisonTree->getNode()->getBranchDestination()->getNode()->getBlock();
       arrayDirectAccessBlock->getExit()->insertTreeTopsAfterMe(directAccessBlock->getEntry(),
-      directAccessBlock->getExit());
+                                                               directAccessBlock->getExit());
+
       cfg->addNode(directAccessBlock);
-      cfg->addEdge(TR::CFGEdge::createEdge(directAccessBlock,  joinBlock, trMemory()));
+      cfg->addEdge(TR::CFGEdge::createEdge(directAccessBlock, joinBlock, trMemory()));
       }
    else
       {
-      directAccessBlock = nullComparisonBlock->getNextBlock();
-      indirectAccessBlock = nullComparisonTree->getNode()->getBranchDestination()->getNode()->getBlock();
+      directAccessBlock = beforeCallBlock->getNextBlock();
```

Comment:
Minor cleanup : This is the first statement in both the if and the else side. So maybe it can be pulled outside the if/then/else